packages/contracts-bedrock/src/L1/SuperchainConfig.sol

```diff
@@ -92,4 +119,49 @@ contract SuperchainConfig is Initializable, ISemver {
         Storage.setAddress(GUARDIAN_SLOT, _guardian);
         emit ConfigUpdate(UpdateType.GUARDIAN, abi.encode(_guardian));
     }
+
+    /// @notice Adds a new chain to the dependency set.
+    ///         Adds the new chain as a dependency for all existing chains in the dependency set, and vice versa. It
+    ///         also stores the SystemConfig address of it, and authorizes the OptimismPortal on the SharedLockbox.
+    /// @param _chainId     The chain ID.
+    /// @param _systemConfig The address of the SystemConfig contract.
+    function addChain(uint256 _chainId, address _systemConfig) external {
+        // TODO: Updater role TBD, using guardian for now.
+        if (msg.sender != guardian()) revert Unauthorized();
+        // Add to the dependency set and check it is not already added (`add()` returns false if it already exists)
+        if (!_dependencySet.add(_chainId)) revert ChainAlreadyAdded();
+
+        systemConfigs[_chainId] = _systemConfig;
+
+        // If the dependency set is empty, there is no need to update the dependencies
+        if (_dependencySet.length() != 0) {
+            // Loop through the dependency set and update the dependency for each chain. Using length - 2 to exclude the
```

Comment:
`length - 2 ` is no longer used

Reply:
`length - 1` is used, and if the length is `0` it will revert